github/repos_rules.go

```diff
@@ -55,14 +60,28 @@ func (s *RepositoriesService) GetRulesForBranch(ctx context.Context, owner, repo
 	return rules, resp, nil
 }
 
+// RepositoryListRulesetsOptions specifies optional parameters to the
+// RepositoriesService.GetAllRulesets method.
+type RepositoryListRulesetsOptions struct {
+	// IncludesParents indicates whether to include rulesets configured at the organization or enterprise level that apply to the repository.
+	IncludesParents *bool `url:"includes_parents,omitempty"`
```

Comment:
According to https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-all-repository-rulesets, the default for `includes_parents` is `true`. So if the user needs to set this to `false`, the `omitempty` will remove the zero-value from the query parameters, so this won't work. It needs to be:

```suggestion
	IncludesParents bool `url:"includes_parents"`
```

Reply:
Yes, you are correct, this is why I put a pointer there so omitempty will not ignore the `false` value. If I don't put a pointer there and remove `omitempty`, `IncludesParents` will always be `false` if the user doesn't explicitly set that field to true and just fills up the pagination. This also references your comment below, because with pointer and `omitempty`,  if the user adds opts, they will not need to explicitly set IncludesParents correctly. It will be just ignored in the query params, which means the Github API will treat it as `true` for `includes_parents`

Reply:
Please let me know what you think about it, in the case of leaving things how they are, with pointer and `omitempty`, the empty value for pointer is nil, which means if the user doesn't fill up that field, it will be omitted, whereas if they want to change the default `true` behaviour of the API, they can set the field to a `Ptr(false)` value, which will not be omitted since it's not an empty value for the pointer.
If I change it to be without a pointer and without `omitempty`, there can occur a different misleading situation:
```go
IncludesParents bool `url:"includes_parents"`
```
In this case, if the user only fills up the pagination and doesn't fill the `IncludesParents` field here, it will always be false. Maybe the user expects it to be `true` since that's what the API docs state, and doesn't fill up this field, but leaving it empty will actually set it to the empty value for bool, which is `false`, when the user expected it to be `true` by default, that may cause confusion

Reply:
Yes, you are correct, this is why I put a pointer there so omitempty will not ignore the false value. If I don't put a pointer there and remove omitempty, IncludesParents will always be false if the user doesn't explicitly set that field to true and just fills up the pagination. This also references your comment below, because with pointer and omitempty, if the user adds opts, they will not need to explicitly set IncludesParents correctly. It will be just ignored in the query params, which means the Github API will treat it as true for includes_parents

Reply:
Please let me know what you think about it, in the case of leaving things how they are, with pointer and `omitempty`, the empty value for pointer is nil, which means if the user doesn't fill up that field, it will be omitted, whereas if they want to change the default `true` behaviour of the API, they can set the field to a `Ptr(false)` value, which will not be omitted since it's not an empty value for the pointer.
If I change it to be without a pointer and without `omitempty`, there can occur a different misleading situation:
```go
IncludesParents bool `url:"includes_parents"`
```
In this case, if the user only fills up the pagination and doesn't fill the `IncludesParents` field here, it will always be false. Maybe the user expects it to be `true` since that's what the API docs state, and doesn't fill up this field, but leaving it empty will actually set it to the empty value for bool, which is `false`, when the user expected it to be `true` by default, that may cause confusion

Reply:
I apologize for the incorrect statements above.  I see what you are saying.